### PR TITLE
Add request mirroring doc

### PR DESCRIPTION
**Pull request by AliceProxy**
Adds user-doc for request mirroring via HTTPRoutes

### CODE REVIEW

File: docs/latest/user/http-request-mirroring.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+Get the Gateway's address:
+
+```shell
+export GATEWAY_HOST=$(kubectl get gateway/eg -o jsonpath='{.status.addresses[0].value}')
+```
+
+Querying `backends.example/get` should result in a `200` response from the example Gateway and the output from the
+example app should indicate which pod handled the request. There is only one pod in the deployment for the example app
+from the quickstart, so it will be the same on all subsequent requests.
+
```
Comment (arkodg) [inline docs/latest/user/http-request-mirroring.md:124]: any way to highlight that `backend-2` received the request ?

File: docs/latest/api/deprecation-policy.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Envoy Gateway API Deprecation Policy
```
Comment (arkodg) [inline docs/latest/api/deprecation-policy.md:1]: can we split this up in a separate PR

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]
